tests/30rooms/60version_upgrade.pl

```diff
@@ -408,6 +408,58 @@ sub upgrade_room_synced {
    };
 
 
+test "/upgrade copies ban events to the new room",
+   requires => [
+      local_user_and_room_fixtures(),
+      qw( can_upgrade_room_version ),
+   ],
+
+   do => sub {
+      my ( $creator, $room_id ) = @_;
+      my ( $new_room_id );
+
+      my $content = {
+         membership => "ban",
+      };
+
+      matrix_put_room_state(
+         $creator, $room_id,
+         type => "m.room.member",
+         content => $content,
+         state_key => '@bob:matrix.org',
+      )->then( sub {
+         matrix_sync( $creator );
+      })->then( sub {
+         upgrade_room_synced(
+            $creator, $room_id,
+            new_version => $TEST_NEW_VERSION,
+         );
+      })->then( sub {
+         ( $new_room_id, ) = @_;
+
+         matrix_sync_again( $creator );
+      })->then( sub {
+         my ( $sync_body ) = @_;
+
+         log_if_fail "sync body", $sync_body;
+
+         my $room = $sync_body->{rooms}{join}{$new_room_id};
+
+         my $event = first {
+            $_->{type} eq "m.room.member" && $_->{state_key} eq '@bob:matrix.org',
+         } @{ $room->{timeline}->{events} };
```

Comment:
We should probably check if its in the `state` section, as that's also a valid place.

On top of that, we can write this entire thing using `await_sync_timeline_contains`:

```perl
await_sync_timeline_contains( $user, $room, check => sub {
    return unless $event->{type} eq "m.room.member";
    return unless $event->{state_key} eq "@bob:matrix.org";

    assert_deeply_eq(
        $event->{content},
        $content,
        "ban in replacement room",
    );

    return 1;
});
```

Which makes the test a lot shorter, and stops you from having to do the manual syncs.

We will have to create a new `await_sync_timeline_or_state_contains` function, but that is fairly trivial to do.